src/main/java/com/stripe/model/CreditNote.java

```diff
@@ -0,0 +1,238 @@
+// Generated by com.stripe.generator.entity.SdkBuilder
+
+package com.stripe.model;
+
+import com.google.gson.annotations.SerializedName;
+import com.stripe.Stripe;
+import com.stripe.exception.StripeException;
+import com.stripe.net.ApiResource;
+import com.stripe.net.RequestOptions;
+import java.util.Map;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.Setter;
+
+@Getter
+@Setter
+@EqualsAndHashCode(callSuper = false)
+public class CreditNote extends ApiResource implements HasId, MetadataStore<CreditNote> {
+  /** Credit note amount. */
+  @SerializedName("amount")
+  Long amount;
+
+  /** Time at which the object was created. Measured in seconds since the Unix epoch. */
+  @SerializedName("created")
+  Long created;
+
+  /**
+   * Three-letter [ISO currency code](https://www.iso.org/iso-4217-currency-codes.html), in
+   * lowercase. Must be a [supported currency](https://stripe.com/docs/currencies).
+   */
+  @SerializedName("currency")
+  String currency;
+
+  /** ID of the customer. */
+  @SerializedName("customer")
+  @Getter(lombok.AccessLevel.NONE)
+  @Setter(lombok.AccessLevel.NONE)
+  ExpandableField<Customer> customer;
+
+  /** Unique identifier for the object. */
+  @Getter(onMethod = @__({@Override}))
+  @SerializedName("id")
+  String id;
+
+  /** ID of the invoice. */
+  @SerializedName("invoice")
+  @Getter(lombok.AccessLevel.NONE)
+  @Setter(lombok.AccessLevel.NONE)
+  ExpandableField<Invoice> invoice;
+
+  /**
+   * Has the value `true` if the object exists in live mode or the value `false` if the object
+   * exists in test mode.
+   */
+  @SerializedName("livemode")
+  Boolean livemode;
+
+  /** Credit note memo. */
+  @SerializedName("memo")
+  String memo;
+
+  /**
+   * Set of key-value pairs that you can attach to an object. This can be useful for storing
+   * additional information about the object in a structured format.
+   */
+  @Getter(onMethod = @__({@Override}))
+  @SerializedName("metadata")
+  Map<String, String> metadata;
+
+  /** Credit note number. */
+  @SerializedName("number")
+  String number;
+
+  /** String representing the object's type. Objects of the same type share the same value. */
+  @SerializedName("object")
+  String object;
+
+  /** The link to download the PDF of the credit note. */
+  @SerializedName("pdf")
+  String pdf;
+
+  /**
+   * Reason for issuing this credit note, one of `duplicate`, `fraudulent`, `order_cancellation`,
+   * `order_change`, `other`, or `product_unsatisfactory`.
+   */
+  @SerializedName("reason")
+  String reason;
```

Comment:
This is not correct because it hasn't been updated per the latest API review yet.